File: Assessment/Assessment.WEB/src/app/services/answer.service.ts
import { Inject } from '@angular/core';
import { Http } from '@angular/http';
import { Answer } from '../entities/answer';

export class AnswerService {
    http: Http;
    baseUrl: string;
    controller: string = "api/Answer/";

    constructor(http: Http, baseUrl: string) {
        this.http = http;
        this.baseUrl = baseUrl;
    }

    public getAll() {
        return this.http.get(this.baseUrl + this.controller);
    }

    public get(id: number) {
        return this.http.get(this.baseUrl + this.controller + id);
    }

    public update(answer: Answer) {
        return this.http.put(this.baseUrl + this.controller, answer);
    }

    public insert(answer: Answer) {
        return this.http.post(this.baseUrl + this.controller, answer);
    }

    public delete(id: number) {
        return this.http.delete(this.baseUrl + this.controller + id);
    }
}

File: Assessment/Assessment.Entities/AnswerService.cs
using System;
using System.Collections.Generic;
using Assessment.Entities.DataInterface;

namespace Assessment.Entities
{
    public class AnswerService
    {
        private IAnswerRepository _repository;

        public AnswerService(IAnswerRepository repository)
        {
            _repository = repository;
        }

        public void Insert(Answer answer)
        {
            _repository.Insert(answer);
        }

        public void Delete(int id)
        {
            _repository.Delete(id);
        }

        public void Update(Answer answer)
        {
            _repository.Update(answer);
        }

        public Answer Get(int answerID)
        {
            return _repository.Get(answerID);
        }

        public IEnumerable<Answer> GetAll()
        {
            return _repository.GetAll();
        }
    }
}